### ISSUE TITLE
2/14 Nov 8 SIM UI/Psy Model: Sankey Diagram Not Portraying Data Correctly

### ISSUE THREAD

**Issue by jamesmrollins** [lzim/teampsd#797]
**1. Add description:**  The Psy Sankey diagram is not portraying variable data the same as other charts in the Expanded Outputs section.

<< Paste screenshots here>>
 
**2. Click on the right:**
a. **Projects** - assign to **issue_tracker.**
b. **Assignee** - assign relevant team members.
c. **Labels**- review and select all potential _MTL_ dependencies.
d. **Milestones** - select the dependent milestone.

-----------------------------------------------------------------------------------------
**Workgroups leads**
**Identify Constraints (check all that apply):**
- [ ]   Capacity
- [ ]   Cost
- [ ]   Schedule
- [ ]   Interference
- [ ]   Capability
- [ ]   Other (describe)

_Add estimated person-hours to complete:_

**Edit the due date (if necessary)**

**Comment by jamesmrollins** [comment]
The following is transposed from an email provided by @TomRust 

Hi gang!

TLDR: The relationship between a rate (visits completed per week) and an accumulation (total visits over patients’ duration) is hard to intuit.  Even for me!  :)  I completed my review of the Sankey, and it looks all good.  It took me too long to review, though, so I’ve added more documentation to clarify some of the Sankey Bars equations; a new PSY version attached.    

Note for Facilitators: I looked through the Sankey formulas after being confused by how Team 2 could both split their time basically 50/50 between patients in their first 3 months and those in care longer than 3 months AND to  have patients in care for >3months consume WAY more visits over their time in care.  But, now that I’ve dug though everything, this makes sense, as their patients in care >3 months are in care for WAY longer than 3 months, so even if the rate is the same (~25 appointments completed per week), being in care for 12 weeks vs ~50 weeks makes a huge difference (especially when a perfect world would look like 12 weeks and 0 weeks after a patient’s first three months!).  Now that I have (perpetually) re-learned the difference between a rate and an accumulation, the surprising thing is not that the total appointments after 3 months for Team 2 is so large, but that it is smaller than I would guess.  If patients are in care for an additional year (4x longer than 3 months), then that accumulation of completed appointments should also be 4x more than the total number of appointments competed in patients’ first three months. Since this isn’t the case, then the average RVI for patients who return after 3 months must be less than when they were in their first three months in care.  

@jamesmrollins  I couldn’t see clearly in the MasterCrosswalk which variables were used for the x-axis in the Sankey bars. Which column has that info? It looks like these variables, but it would be great to double-check:

Visit count in BC Sankey for Team 2 data “640a0_team2_2019_08_01.xlsx” (i.e., x-axis of Sankey Bars)
26 visits = Vensim variable “H-Bars” = total visits completed per patient (including visits in their first 3 months in care) who follow the “Complete and Continue” path
17 visits = Vensim variable “E-Bars” = total visits completed per patient (including visits in their first 3 months in care) who follow the “Initiate and Return” path
9 visits = Vensim variable “C-Bars” = total visits completed per patient (including visits in their first 3 months in care) who follow the “Start and Return” path
12 visits = Vensim variable “G-Bars” = total visits completed per patient who follow the “Complete and Graduate” path
2 visits = Vensim variable “D-Bars” = total visits completed per patient who follow the “Initiate, but Quit Early” path
1 visit = Hardcoded to 1, but we could use Vensim variable “Bars E3” = total visits completed per patient who follow the “1-n-Done” path

**Comment by jamesmrollins** [comment]
@TomRust @lzim @staceypark  I will upload the new model into test.  From there @TomRust can verify as will I.  I will review the master-crosswalk table and update as well.

**Comment by jamesmrollins** [comment]
@lzim @TomRust the new model is in the TEST environment. There is a pull request pending to update the model repository.  Crosswalk table scrub is pending.  Will notify when complete.

FYI @staceypark @ritahitching

**Comment by staceypark** [comment]
Note: @jamesmrollins will be able to promote just this piece to PROD without needing to promote other items in TEST as well, as it is specifically the model that needs updating.

**Comment by lzim** [comment]
Thanks all!

I just approved the pull request. But, not sure I'm tracking everything.

Just checking: @TomRust and I will be walking through PSY tomorrow for our workshop, which means some folks will access PSY in mtl.how/demo. Tom and I will be working this afternoon at mtl.how/sim PROD.

Are we ready? Thanks!

**Comment by jamesmrollins** [comment]
@lzim @TomRust @staceypark @ritahitching the newest Psy model is NOT updated in PROD.  It is still in test.  Did you and Tom want to evaluate in TEST tomorrow, or did you just want me to promote to PROD and DEMO to support your workshop?

**Comment by lzim** [comment]
Thanks @jamesmrollins 

@TomRust says we do NOT need to promote before the workshop tomorrow.

Right Tom?

FYI: @staceypark and @ritahitching

**Comment by staceypark** [comment]
@jamesmrollins is this still in TEST? or has it been promoted to PROD?

**Comment by jamesmrollins** [comment]
@staceypark this is in test.

**Comment by staceypark** [comment]
@jamesmrollins I noticed you cross-referenced this in #949 as a sub-item to be completed together by 2/14. I updated the due date to match.

**Comment by lzim** [comment]
> @jamesmrollins I noticed you cross-referenced this in #949 as a sub-item to be completed together by 2/14. I updated the due date to match.

@staceypark 
Hi, what’s going on this issue? I’m not sure I’m tracking it above. 
But, I assigned it to **teams_sim** so that it’s not invisible to us!

**Comment by staceypark** [comment]
@lzim @jamesmrollins @TomRust 
I believe the work was done for this awhile ago. 
1. Was it ever validated by Tom?
2. And is this in TEST or PROD right now?

**Comment by staceypark** [comment]
As discussed at 3/11 support workgroups meeting, @jamesmrollins will follow up on this with Tom this Friday 3/13

**Comment by TomRust** [comment]
Hi All, 

The "new" version of PSY that is in TEST that triggered this thread back in November is basically the same as the version in PROD -- all I did was improve the documentation that goes with the Sankey variables. It can be promoted to PROD.

My read on this thread is that it is not actually a question about the model, but about the master crosswalk.  I just needed James to double-check that the data were being pulled into the Sankey Bars chart the right way.  They appear to be (by cross checking the values in the Vensim against the sankey bars displayed in the Sim UI), but I couldn't be 100% sure by seeing the data links written out in the master crosswalk.

My recollection is that there are six values to check for the Sankey Bars:
So, is the ... 
1) Vensim variable “H-Bars” = total visits completed per patient (including visits in their first 3 months in care) who follow the “Complete and Continue” path?
2) Vensim variable “E-Bars” = total visits completed per patient (including visits in their first 3 months in care) who follow the “Initiate and Return” path?
3) Vensim variable “C-Bars” = total visits completed per patient (including visits in their first 3 months in care) who follow the “Start and Return” path?
4) Vensim variable “G-Bars” = total visits completed per patient who follow the “Complete and Graduate” path?
5) Vensim variable “D-Bars” = total visits completed per patient who follow the “Initiate, but Quit Early” path?
6) Vensim variable “Bars E3” = total visits completed per patient who follow the “1-n-Done” path, or is it hard coded to 1 (both are fine)?

Sorry for being so slow to respond!!!!!

**Comment by jamesmrollins** [comment]
@TomRust Here is the Sankey chart variable map.


[SankeyChartVariable.docx](https://github.com/lzim/teampsd/files/4340812/SankeyChartVariable.docx)

**Comment by staceypark** [comment]
We needed to verify that we were correctly mapped between the Vensim and the Sim.
@TomRust just reviewed @jamesmrollins attached doc and confirmed everything looks correct.

This issue has already been promoted to PROD in _MTL_ 2.0

As such, we are ready to close this issue barring further questions.

> @TomRust Here is the Sankey chart variable map.
> 
> [SankeyChartVariable.docx](https://github.com/lzim/teampsd/files/4340812/SankeyChartVariable.docx)
